stingray/lombscargle.py

```diff
@@ -0,0 +1,643 @@
+                    normalization="psd",
+                ).autofrequency(
+                    minimum_frequency=max(self.min_freq, 0), maximum_frequency=self.max_freq
+                ),
+            )[0]
+            if max(freqs2) > max(freq):
+                freq = freqs2
+
+        if method == "slow":
+            lsft1 = lsft_slow(lc1.counts, lc1.time, freq)
+            lsft2 = lsft_slow(lc2.counts, lc2.time, freq)
+        elif method == "fast":
+            lsft1 = lsft_fast(lc1.counts, lc1.time, freq, oversampling=oversampling)
+            lsft2 = lsft_fast(lc2.counts, lc2.time, freq, oversampling=oversampling)
+        if fullspec:
+            lsft1, _ = impose_symmetry_lsft(lsft1, np.sum((lc1.counts)), lc1.n, freq)
+            lsft2, freq = impose_symmetry_lsft(lsft2, np.sum(lc2.counts), lc2.n, freq)
+        cross = np.multiply(lsft1, np.conjugate(lsft2))
+        return freq, cross
+
+    def _initialize_empty(self):
+        self.freq = None
+        self.power = None
+        self.power_err = None
+        self.unnorm_power = None
+        self.unnorm_power_err = None
+        self.df = None
+        self.dt = None
+        self.nphots1 = None
+        self.nphots2 = None
+        self.m = 1
+        self.n = None
+        self.fullspec = None
+        self.k = 1
+        self.err_dist = None
+        self.method = None
+        self.meancounts1 = None
+        self.meancounts2 = None
+        self.oversampling = None
+        self.variance1 = None
+        self.variance2 = None
+        return
+
+    def phase_lag(self):
+        """Not applicable for unevenly sampled data"""
+        raise AttributeError(
+            "Object has no attribute named 'phase_lag' ! Not applicable for unevenly sampled data"
+        )
+
+    def time_lag(self):
+        """Not applicable for unevenly sampled data"""
+        raise AttributeError(
+            "Object has no attribute named 'time_lag' ! Not applicable for unevenly sampled data"
+        )
+
+    def classical_significances(self):
+        """Not applicable for unevenly sampled data"""
+        raise AttributeError(
+            "Object has no attribute named 'classical_significances' ! Not applicable for unevenly sampled data"
+        )
+
+    def from_time_array(self):
+        """Not applicable for unevenly sampled data"""
+        raise AttributeError(
+            "Object has no attribute named 'from_time_array' ! Not applicable for unevenly sampled data"
+        )
+
+    def from_events(self):
+        """Not applicable for unevenly sampled data"""
+        raise AttributeError(
+            "Object has no attribute named 'from_events' ! Not applicable for unevenly sampled data"
+        )
+
+    def from_lightcurve(self):
+        """Not applicable for unevenly sampled data"""
+        raise AttributeError(
+            "Object has no attribute named 'from_lightcurve' ! Not applicable for unevenly sampled data"
+        )
+
+    def from_lc_iterable(self):
```

Comment:
They actually don't throw any error since I am inheriting from Crossspectrum and the parent methods dont have any filter except time_lag. Which is anyway a valid method so it needs to be overridden anyway to surpass it.